### ISSUE TITLE
stop tech leakage

### ISSUE THREAD

**Issue by bvanevery** [tnevolin/thinker-doer#92]
I've now played all factions in the 25X versions of WTP.  Various game systems have come a long way.  I think early game mindworm stuff is fine, seems balanced.  Combat system and unit production costs seem well in hand.  However there are some things I definitely consider a problem.

Tech is ridiculously cheap.  There is no reason to do any research at all, as you will simply buy all your tech for next to nothing, from factions that freely sell it.  Or even give it away to allies.  RESEARCH penalties that some factions carry, like the Believers and the Free Drones, don't mean a darned thing.  To the extent they are meant to offset and "pay for" their faction bonuses, it's a complete deal / slam dunk.  A disadvantage that means absolutely nothing to them.

Meanwhile, other factions like the University and the Cyborgs, have piles of really serious disadvantages, and their "bonus" of RESEARCH is completely unimportant.  As human players, they will never get ahead of the AI city sprawl.  That level of sprawl, seriously speeds up research.

In the stock game, factions refuse to trade techs for which Secret Projects have not been completed.  There are some exceptions to this, which tend to come across as buggy behavior, usually on the part of the Cyborgs and the Data Angels.  I was consistently surprised that they'd sell me stuff that they really shouldn't.  But pretty much all the other factions, if a SP isn't done, you're not getting the tech.  End of story.  You can flip your diplomatic dialog options 'till you're blue in the face, and you'll be completely wasting your time.

In my mod, I successfully use this behavior to put the brakes on faction tech trading.  Important stuff, I bottle up with a SP.  In WTP you only see this behavior in a few pathological instances.  I'm not sure why they think the Human Genome Project is so valuable, but it does keep the Recycling Tanks from getting traded too easily.  I've gone a long time not being able to acquire that from anyone else.  They're also super cagey about the Universal Translator.  I don't know why they're not similarly hung up on the Merchant Exchange, which costs the same low number of minerals.  It doesn't seem to be about the minerals.

I don't know if original tech trading code was being used, or if it got turned into C++.  I am wondering if the "don't trade SPs" code was deleted, or just altered, or its input values changed enough to alter its in practice behavior.  But something is not right.  It harms the game a great deal.

Another way to stop tech leakage, is to have the AIs either refuse to trade techs much more often, or have them charge a great deal of money for stuff.  So that you can't just count on your ally of convenience, offering you the next tech for 100 credits.  I have no idea how difficult it is to change the code for this though.  If it's one of these "find the ASM" things, dissect it, and write something totally new in C++, I'm not signing up for the job.

But I do think it is important to track this issue, and deal with it somehow.  It's partly ruining the game.  2 factions are getting a cakewalk, and 2 factions are pretty much useless.

**Comment by tnevolin** [comment]
Don't know what could be done in this regard. Reduce factions willingness to trade techs? Increase risk of stealing tech?

**Comment by bvanevery** [comment]
Never trading techs that have incomplete Secret Projects, would help.  It's the path of least resistance and almost always the stock game behavior.

Increasing the risk of probe teams won't help.  Players or AI will just send more probe teams.

**Comment by tnevolin** [comment]
> Never trading techs that have incomplete Secret Projects, would help.

Nah. This is too narrow tailored. They already don't trade such tech. I meant to reduce trading in general.

> Increasing the risk of probe teams won't help. Players or AI will just send more probe teams.

Err, **just**? If one need 100 time more probe teams to steal a tech that would certainly reduce acquisition frequency together with significant increase in cost.

**Comment by bvanevery** [comment]
> They already don't trade such tech. 

This is false in many cases.  I regularly acquire SP viable techs from trades.  There seem to be some SPs that the factions fixate on not trading, and others they don't care about.

>  If one need 100 time more probe teams to steal a tech 

It will dissuade human players due to tedium.  But not AIs because they are immune to tedium.  They'll just make more probes, and one will get through.  AIs can bang out all kinds of units pretty quickly.  They aren't operating on the same level playing field as a human player, nor making as much judgment about what to produce.

**Comment by tnevolin** [comment]
I've created topic on reddit to solicit options. Let's see how it goes.

**Comment by bvanevery** [comment]
I'm going to refrain from jumping in on the discussion, until / unless others get into it.  I've seen an awful lot of crickets chirping when you've asked about aspects of your work.  It's not easy to build an audience for an old game, let alone for one of a few mods of it.

To me the puzzle seems to require a commercial solution.  Although, I wonder even then.  I haven't actually spent time on any specific 4X game forums lately, and honestly, maybe I never have.  Talking about SMAC was like my 1st effort, and it did _not_ go well with the multiplayer crowd.  I probably ought to study the communities of various commercial games, to see if any of them have people who say things that are worth hearing.

**Comment by tnevolin** [comment]
Yes. There are a lot of irrelevant opinions but there are relevant too. I'll filter.

**Comment by tnevolin** [comment]
The response on reddit clearly states that research is a powerful bonus. I tend to agree since in my practical experience it is absolute technical advantage matters not relative. And faction with higher research will always be few tech ahead thus keeping their advantage throughout the game even if slower factions keep catching up all the time.
In other words penalizing slow research factions even more will make their lag grow proportionally throughout the game so they will be an absolute outlawyer after mid game. Meaning it would not be possible for them to win.
People there also made a perfect point that University and Cyborgs are perceived best factions whereas Believers are the worst. It may not be an exact case but it should be close to truth. Otherwise, people would not chose these factions.

I also agree that University AI sucks because it does not capitalize on their inventions. Whereas University human rocks exactly by this reason. Nothing can be easily done here. Let's live with this for now.

Another reason I don't want to go there is that any changes in game concept like these are huge and benefit is not clearly seen until tried. I don't want to spend my time just to experiment on that and see if we can find a sweet spot at all. There is a certain pain threshold that I am willing to cross only when current functionality is irrefutably unacceptable.

**Comment by bvanevery** [comment]
_My_ opinion is current functionality is irrefutably unacceptable, despite you having closed this issue.  Allies will simply give you every tech imaginable.  You don't have to do any research at all.  They will give you very, very advanced tech.  From [my last post in your dev thread](https://alphacentauri2.info/index.php?topic=21359.msg132876#msg132876) before quitting in disgust:

> Ok look I'm quitting.  The game's level of tech leakage is ridiculous.  I haven't researched a darned thing.  Zak wanted war with Deirdre, I declined.  Zak then allies with me anyways.  I get free tech from him.
>
>Then I talk to Lal and he trades me a bunch of stuff, that previously he wasn't interested in trading.  So now I'm just loaded.  I haven't researched a darned thing, and you're obviously a complete chump to do any research in this game.  It's totally out of balance and it makes a lot of pace of the game pointless.
>
>Why do I need to worry about Wealth and Free Market, when Eudaimonic is right around the corner?   Well, maybe it isn't, but I've got a really strong feeling that my own tech choices in this game are completely meaningless.  I'm going to be given everything, which makes this pretty much a baby game.
>
>I know I've filed this in the Issue Tracker before, and it's time to ping that if it hasn't been pinged recently.  This is just crazy.
[Morgan of the Morganites, 2261.zip](https://github.com/tnevolin/thinker-doer/files/8004774/Morgan.of.the.Morganites.2261.zip)

As to why Redditors have wrong opinions about RESEARCH bonus in **your mod**, I think it's because they give their opinion about RESEARCH in plain SMAC, and not your mod.  That's my memory of how that thread went, and I said so publicly at the time.

**Comment by tnevolin** [comment]
I closed it not because it is acceptable but because I don't see how I can do it properly and consistently. Let's reopen it and keep it as a reminder.

Do you think human benefits from this more than AI? Would it be right to employ some anti-human measures or we need to restrict everybody equally?

**Comment by bvanevery** [comment]
What I've observed over the course of many games, is certain factions are research powerhouses, and others are not.  For instance, I can count on the Gaians, the Peacekeepers, the University, and the Cyborgs to be major tech bestowers.  That's just off the top of my head.  The first 3 of those factions were in my last game.  2 became allies, and 1 traded tech totally without restraint.  I don't know if they're doing this with other AIs, but with the human player, they're just giving away the farm with no restraint at all.

The Usurpers, in contrast, are not good at research.  They filled in 1 tech early on, an important one, Fungicidal Tanks.  Otherwise though, they were way behind the powerhouse factions on tech.  It almost makes me wonder if 1 set of factions builds Network Nodes and another doesn't.  The Spartans are bad at tech too.  I don't remember the Hive being particularly good.  Think the Free Drones were pretty bad in my last game, even given their stupendous size and productivity.

Do the AIs even trade techs with each other?  I've sometimes gotten the impression, that they simply don't.  Steal, yes, but trade, I've not seen much evidence for that.  If they were trading, they should all have parity.  Even the powerhouse factions.  Yet, they tend to have different tech offerings.

It is not appropriate to trade techs with unfinished SPs, just because SPs take a long time to build.  I've consistently proven that I can run circles around the current AI in the SP races.  All you have to do is build a vertical city, which the AI is currently incompetent at.  You'll finish any given SP first.  The number of SPs you can finish before the AI does, mainly depends on how many vertical cities you manage to cough up.  Hence the AI tends to do better with SP races earlier when you're still struggling to get your vertical stuff done.  Once you do though, after a certain number of cities the AI has no chance.  It makes for long, boring games, but it's very consistent as to results.  AI can't hang.

AI should flat out refuse to trade techs with unfinished SPs, like in the original game.  AI is currently only doing it for a few early SPs.  Once you blow past those somehow, like by finding a faction that isn't as picky about it, or just researching them yourself, the AI basically gives up as far as trying to restrain your tech advancement.  It will give you everything, and if an ally, at dirty cheap prices like 100 credits.  Factions that like you less, I've seen them charge stiffer prices, which I've often been unwilling to pay.  If only because I don't really need the techs and know I can probably get them from allies soon enough anyways.

AI should also be aware of dangerous techs, like Doctrine: Air Power.

**Comment by tnevolin** [comment]
Of course, they trade. If you turn on the omniscient view and start the game you'll quite often see popups: faction A gets tech from faction B.

OK, noted about SP tech. May need to look at this. I don't think I have changed anything on that yet. So it should be as in original.

**Comment by tnevolin** [comment]
#297
Check it out. Is this sufficient hindering to tech exchange?

**Comment by bvanevery** [comment]
As per my original statement of the Issue, it's about tech _trading_, not stealing.  Please don't make it gratuitously impossible to steal techs.  It's a fundamental game mechanic and it's actually work to steal techs, particularly over long distances on larger maps.  Whereas, being given cheap _trades_ or so many outright gifts, is the big problem.

**Comment by tnevolin** [comment]
Got. It removed stealing ban. Just reduced trading frequency.

**Comment by tnevolin** [comment]
Added few things:

- [ ] Reduced tech trade likelihood.
- [ ] Increased number of factions required to possess tech to be acquired via The Planetary Datalinks to 5.
- [ ] The TECHSHARE and SHARETECH are specialized for faction and can be tuned up individually. Did not modify them.

**Comment by bvanevery** [comment]
To 5 players, waaat??  There's only 7 players in the game!  You're one of 'em.  That means "unknown to only 1 faction other than yourself".  Meaning, probably someone will just sell it to you as very common old tech.

**Comment by tnevolin** [comment]
And? What is wrong with that?
What do you want it to be? 4?

**Comment by bvanevery** [comment]
Based on my extensive playtesting of my own mod, I don't want you to change the number of factions for the Planetary Datalinks at all.  I suggest you revert that change.  You are making way too many game changes at once that affect tech rate.  Any one of them could be sufficient to fix a problem, but it's like you're taking a machete to a jungle.  Just slow down and actually test these changes one at a time.

I also think you've priced Planetary Datalinks way too high for what it does.  It shouldn't cost more than a Command Nexus.  Even if you fix all tech leakage and pacing issues, that would still be true.

**Comment by tnevolin** [comment]
It already costs as Command Nexus.
I will revert changes.

**Comment by bvanevery** [comment]
Ok, I was not aware of the change to be equal to CN.  IMO keeping PD at 3 factions is the only clearly necessary reversion.  Other changes should be tested.

**Comment by tnevolin** [comment]
Right. I mean reverting PD to 3 factions. Other changes are still there.
